src/core/bittorrent/torrenthandle.cpp

```diff
@@ -1203,7 +1203,11 @@ void TorrentHandle::setFirstLastPiecePriority(bool b)
 
             QPair<int, int> extremities = fileExtremityPieces(index);
             SAFE_CALL(piece_priority, extremities.first, prio);
-            SAFE_CALL(piece_priority, extremities.second, prio);
+            // worst case: AVI index = 1% of total file size
+            int nLastPieces = ceil(fileSize(index) * 0.01 / pieceLength());
+            for (int i = 0; i < nLastPieces; ++i)
+                SAFE_CALL(piece_priority, extremities.second - i, prio);
```

Comment:
Better first extract the current priorities in vector using ```torrent_handle::piece_priorities()```, edit it, and then use ```torrent_handle::prioritize_pieces()``` to set them all at once.